dfencoder/autoencoder.py

```diff
@@ -945,26 +945,46 @@ def get_anomaly_score_with_losses(self, df):
         return mse, bce, cce, net
 
     def get_anomaly_score_losses(self, df):
+        """
+        Run the input dataframe `df` through the autoencoder to get the recovery losses by feature type 
+        (numerical/boolean/categorical). 
+        """
         self.eval()
-        data = self.prepare_df(df)
-        input = self.build_input_tensor(data)
 
+        n_batches = len(df) // self.batch_size
+        if len(df) % self.batch_size > 0:
+            n_batches += 1
+            
+        data = self.prepare_df(df)
         num_target, bin_target, codes = self.compute_targets(data)
-
+        # data structures to store the results for each batch
+        # resutls of each categorical feature  are stored separately
+        num_slices, bin_slices, cats_slices = [], [], [[] for _ in self.categorical_fts] 
```

Comment:
Ideally, we would want all of the code to be within the batching loop

Reply:
Addressed in the[ latest commit](https://github.com/nv-morpheus/dfencoder/pull/5/commits/0dd3d60cd3586274211cc9856b1f839696f8bfed)